lib/ansible/module_utils/facts/hardware/sunos.py

```diff
@@ -168,8 +168,15 @@ def get_mount_facts(self):
     def get_dmi_facts(self):
         dmi_facts = {}
 
-        uname_path = self.module.get_bin_path("prtdiag")
-        rc, out, err = self.module.run_command(uname_path)
+        prtdiag_path = self.module.get_bin_path("prtdiag")
+
+        # On Solaris 8 the prtdiag wrapper is absent from /usr/sbin,
+        # but that's okay, because we know where to find the real thing:
+        if not prtdiag_path:
+            rc, platform, err = self.module.run_command('/usr/bin/uname -i')
+            prtdiag_path = '/usr/platform/' + platform.rstrip() + '/sbin/prtdiag'
```

Comment:
instead of assuming that if not found this path does exist, just pass the search pat as an option to get_bin_path and it will error out if even this fallback is not present

Reply:
Will do.  Fact gathering is already slow, so I was reluctant to add an additional fork and exec of `uname` unless it was required.

Reply:
you could still make it conditional, just call get_bin_path again instead of assuming that the binary exists